src/main-cli.cpp

```diff
@@ -85,35 +113,131 @@ int main(int argc, char **argv)
 	L	"PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF"
 	L	"LIABILITY, WHETHERIN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE"
 	L	"OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF"
-	L	"ADVISEDOF THE POSSIBILITY OF SUCH DAMAGE." << std::endl;
-#undef L
+	L	"ADVISEDOF THE POSSIBILITY OF SUCH DAMAGE."
+	L "" << std::endl;
+	#undef L
+}
+
+void print_presets() {
+	DIR *dir;
+	struct dirent *ent;
+	std::string dirName(DEFAULT_CONFIG_DIR);
+	if ((dir = opendir (dirName.c_str())) != NULL) {
+		while ((ent = readdir (dir)) != NULL) {
+			if (ent->d_name[0]=='.') continue; 
+			std::cout << ent->d_name << std::endl;
+		}
+		closedir (dir);
+	} else {
+		std::cerr << "Could not open presets directory!" << std::endl;
+	}
+}
+
+void process_command(YC20Jack &processor, char *command) {
+	char arg1[128];
+
+	if (strcmp(command, "get_presets\n") == 0) {
+		print_presets();
+	}
+	else if (sscanf(command, "set_preset %[a-zA-Z0-9_:/-.#]\n", arg1) == 1) {
```

Comment:
Here I suggest to have a size check to guard against the buffer overflow on `arg1`.
Also I think `-` is not right, as the manual says: `To include a hyphen, make it the last character before the final close bracket.`
Also do we really want to have `/` in the set, considering `print_presets` doesn't list further than 1 depth level of directories?

Reply:
Maybe the width modifier for the scanf pattern could help? http://www.cplusplus.com/reference/cstdio/scanf/

Reply:
The last commit used it:

`else if (sscanf(command, "set_preset %127[a-zA-Z0-9_:.#&%%()+-]\n", arg1) == 1) {`

Regards,